### PR TITLE
Remove deprecated RGBASM features

### LINKED ISSUE #986: Make the automatic `ld`-to-`ldh` optimization off by default, with a gradual switchover
Currently rgbasm optimizes `ld` to `ldh` when it can, with the `-L` flag disabling that behavior.

In the next release of rgbds (0.6.1), add a do-nothing `-l` flag. Add a deprecation warning on the first time that the `ld`-to-`ldh` optimization occurs, saying that it will be off by default in the next release, and you should then use `-l` to enable it.

In the release after that (0.7.0), change it from on by default to off by default, with `-l` turning it on. (`OPT` should support `l` same as it now supports `L`.) Add a deprecation warning on use of the `-L` flag, saying that it is now useless and will be removed or repurposed in the next release. Passing both `-L` and `-l` should be a fatal error.

In the release after *that* (0.8.0), remove the `-l` and `-L` flags.

We should have a similar transition for the `-h` flag, which stops rgbasm from automatically inserting `nop` after `halt`.

### LINKED ISSUE #1194: Remove previously deprecated things for 0.7.0
- [ ] `\,` inside strings
- [ ] `name: MACRO`
- [ ] `__LINE__`
- [ ] `__FILE__`
- [ ] The next step in #986

### PR DESCRIPTION
**Pull request by Rangi42**
- Escaped commas `"\,"` inside strings
- `name: MACRO` syntax
- `__FILE__` and `__LINE__`
- `-H/--nop-after-halt` and `-l/--auto-ldh` on by default

Fixes #1194
Fixes #986 (`-H` `-h` `-L` `-l` are now ordinary deprecated features and will be removed as usual in the release after 0.7.0)

### CODE REVIEW

File: src/gfx/process.cpp [hunk 1]
```diff
@@ -680,7 +680,11 @@ class TileData {
 		for (uint32_t x = 0; x < 8; ++x) {
 			row <<= 1;
 			uint8_t index = palette.indexOf(tile.pixel(x, y).cgbColor());
-			assert(index < palette.size()); // The color should be in the palette
+			if (!(index < palette.size())) {
+				fprintf(stderr, "index %d >= palette.size() %d!\n",
+					index, palette.size());
+				exit(1);
+			}
```
Comment (ISSOtm) [inline src/gfx/process.cpp:687]: Assertions are only meant as internal consistency checks; why promote this?

File: test/asm/.gitignore [hunk 2]
```diff
@@ -1,3 +1,2 @@
-/quote\"file.*
```
Comment (ISSOtm) [inline test/asm/.gitignore:1]: Doesn't this file have special handling in the test runner script?

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Hopefully that's the final round of deprecations for a long time...

File: src/gfx/process.cpp (hunk 1 above)
Comment (Rangi42) [reply src/gfx/process.cpp:687]: Oops, this is left over from trying to reproduce that rgbgfx bug report.

File: test/asm/.gitignore (hunk 2 above)
Comment (Rangi42) [reply test/asm/.gitignore:1]: That's what I thought, but couldn't find any other mention of it. It's version.asm that has special handling.

Comment (ISSOtm) [reply test/asm/.gitignore:1]: The handling was removed in #1072, actually; this is a leftover.

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**